configure.py

```diff
@@ -305,6 +305,8 @@ def binary(name):
 else:
     n.variable('ar', configure_env.get('AR', 'ar'))
 
+# Note that build settings are separately specified in CMakeLists.txt and
```

Comment:
i think better would be to simply remove this file entirely.

Reply:
I would be happy to do that but it is currently used by the OSX builders, and Linux as well I believe. I don't know how such things are configured.

Reply:
I don't see anything about python in the osx CI config here: 
https://github.com/ninja-build/ninja/blob/master/.github/workflows/macos.yml
I do see "build-with-python" in the linux CI: https://github.com/ninja-build/ninja/blob/master/.github/workflows/linux.yml
Just remove everything after line 129 in that file.

Reply:
I misread and thought you were suggesting that we remove CMakeLists.txt. Removing configure.py seems like a big step. It is referenced in README.md and in https://github.com/ninja-build/ninja/wiki and it is what I have always used to build ninja. I think that removing configure.py is out-of-scope for this PR.

Reply:
> I think that removing configure.py is out-of-scope for this PR.

Indeed.